### PR TITLE
fix: validate percentage fields between 0 and 100

### LINKED ISSUE #4995: Multiple percentage fields accept out-of-range values due to missing/incomplete validators
**Describe the bug**

Several percentage-based fields across settings and forms lack proper bounds validation. Some only reject negative values but accept anything above 100%; others accept neither negative nor over-100% values at all. This mirrors the bug fixed in #4962/#4964 for `default_price`, but for percentage fields instead of prices.

Affected fields:

Category A — has MinValueValidator(0) but no upper bound (accepts >100%):
- `ticket_fee_percentage` — control/forms/global_settings.py (~L417), used in `amount * (rate / 100)` (eventyay_common/tasks.py)
- `payment_stripe_connect_app_fee_percent` — control/forms/global_settings.py (~L368)

Category B — no validators at all (accepts negative *and* >100%):
- `TaxRule.rate` (`tax_rate`) — form: control/forms/event.py (`EventWizardBasicsForm`, ~L182); model: base/models/tax.py:138
- `cancel_allow_user_paid_keep_percentage` — base/configurations/default_setting.py (~L1460), used in base/models/orders.py
- `keep_fee_percentage` — control/forms/orders.py (~L697), used in base/services/cancelevent.py

Category C — conditional percent field, needs a clean()-based check instead of a blanket max:
- `Voucher.value` when `price_mode == 'percent'` — with value=1000, computes a **negative price**, same failure mode as #4962
- `OrderFee.value` + `_treat_value_as_percentage` (API, api/serializers/order.py ~L1527) — same risk

**Expected behavior**
Percentage fields should reject values outside 0–100% (or, for the Category C fields, reject values >100% only when the field is being interpreted as a percentage).

**Screenshots**

Image

**Additional context**
These aren't cosmetic — bad values compute directly into billing invoices, ticket fees, and cancellation refunds. A typo (extra zero) would silently generate a fee/refund many times the correct amount instead of being rejected at input time.

Suggested fix: add `MaxValueValidator(100)` for Category A/B fields (same pattern as #4964), and a conditional `clean()` check for Category C fields (Voucher/OrderFee) since their `value` field is reused across non-percentage modes.

Happy to submit a PR for this if it's welcome.

### PR DESCRIPTION
**Pull request by ankitgup2637-ai**
## Description

Fixes #4995

Several percentage-based fields accepted values outside the valid 0–100% range.

### Changes

- Added 0–100% validation to affected percentage fields.
- Added conditional validation for voucher percentage pricing.
- Added validation for percentage-based order fees.
- Added a migration for the TaxRule rate validation.
- Added regression tests for percentage boundary values.

### Testing

- `tests/tickets/test_percentage_validation.py` — 6 passed
- `git diff --check` — passed
 
 ### video recording

https://github.com/user-attachments/assets/19374b84-b8b2-4d00-9556-5961e492845a

## Summary by Sourcery

Enforce valid 0–100 bounds for all percentage-based configuration and transaction values.

Bug Fixes:
- Restrict percentage-based fields across taxes, fees, cancellation settings, voucher pricing, and order values to the valid 0–100 range.

Enhancements:
- Add conditional validation for percentage-mode voucher and order values.
- Add TaxRule rate validators and the corresponding database migration.

Tests:
- Add regression coverage for TaxRule percentage boundary validation.

### CODE REVIEW

File: app/eventyay/base/migrations/0055_alter_taxrule_rate_and_more.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
+# Generated by Django 6.0.7 on 2026-08-17 12:01
+
+import django.core.validators
+from django.db import migrations, models
+
```
Comment (Rachit7168) [inline app/eventyay/base/migrations/0055_alter_taxrule_rate_and_more.py]: Unrelated to the PR remove this

File: app/eventyay/base/migrations/0056_merge_20260817_2003.py [hunk 2]
```diff
@@ -0,0 +1,14 @@
+# Generated by Django 6.0.7 on 2026-08-17 20:03
+
+from django.db import migrations
+
```
Comment (Rachit7168) [inline app/eventyay/base/migrations/0056_merge_20260817_2003.py]: Make the migration scoped to the PR only

**Review by Rachit7168 — COMMENTED** [review]
Thanks for working on this !

File: app/tests/tickets/test_percentage_validation.py [hunk 3]
```diff
@@ -0,0 +1,136 @@
+import pytest
```
Comment (Aqil-Ahmad) [inline app/tests/tickets/test_percentage_validation.py]: you are over-testing a simple fix. remove unnecessary or redundant tests

**Review by Aqil-Ahmad — CHANGES_REQUESTED** [review]

**Review by Rachit7168 — CHANGES_REQUESTED** [review]
you missed one in the payment provider settings. The _fee_percent field (which represents the percentage fee a payment provider like Stripe or PayPal charges) is missing the validators.

In app/eventyay/base/payment.py, you need to update the _fee_percent field definition

and also renumber migration

**Review by Saksham-Sirohi — CHANGES_REQUESTED** [review]
Validation should not require a migration, please address

**Review by ArnavBallinCode — APPROVED** [review]
This is a focused and well-executed fix for issue #4995.

I've reviewed the architecture and ran the test suite locally. The author successfully addressed all previous reviewer comments:
1. **Missing `_fee_percent` validation**: The `[MinValueValidator(0), MaxValueValidator(100)]` validators were correctly added to `_fee_percent` inside `app/eventyay/base/payment.py`'s `settings_form_fields`.
2. **Avoiding Migrations**: Dropping the migration and moving the validation logic to the `clean()` method of `TaxRule` and `Voucher` models is perfectly aligned with Django best practices. It prevents crashes during deployment against any legacy dirty data while rejecting new invalid inputs.

**API Serializers**: The conditional checks (`price_mode == 'percent'`) are cleanly isolated in the `validate()` methods.
**Form Validators**: The Django validators will correctly enforce native HTML5 boundary checks on the UI.
**Tests**: All 24 tests in `test_taxrules.py` pass without issues, covering the regression cases natively via `full_clean()`.

Everything looks great and the PR is fully production-ready. Approved!

**Review by Saksham-Sirohi — APPROVED** [review]
